doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst

```diff
@@ -0,0 +1,899 @@
+
+.. code-block:: c++
+
+        {
+          auto stage =
+              std::make_unique<mtc::stages::ModifyPlanningScene>("forbid collision (hand,object)");
+          stage->allowCollisions("object",
+                                task.getRobotModel()
+                                    ->getJointModelGroup(hand_group_name)
+                                    ->getLinkModelNamesWithCollisionGeometry(),
+                                false);
+          place->insert(std::move(stage));
+        }
+
+Now, we can detach the object using ``detachObject``.
+
+.. code-block:: c++
+
+        {
+          auto stage = std::make_unique<mtc::stages::ModifyPlanningScene>("detach object");
+          stage->detachObject("object", hand_frame);
+          place->insert(std::move(stage));
+        }
+
+We retreat from the object using a ``MoveRelative`` stage, which is done similarly to the ``approach object`` and ``lift object`` stages.
+
+.. code-block:: c++
+
+        {
+          auto stage = std::make_unique<mtc::stages::MoveRelative>("retreat", cartesian_planner);
+          stage->properties().configureInitFrom(mtc::Stage::PARENT, { "group" });
+          stage->setMinMaxDistance(0.1, 0.3);
+          stage->setIKFrame(hand_frame);
+          stage->properties().set("marker_ns", "retreat");
+
+          // Set retreat direction
+          geometry_msgs::msg::Vector3Stamped vec;
+          vec.header.frame_id = "world";
+          vec.vector.x = -0.5;
+          stage->setDirection(vec);
+          place->insert(std::move(stage));
+        }
+
+We finish our place serial container and add it to the task.
+
+.. code-block:: c++
+
+        task.add(std::move(place));
+      }
+
+The final step is to return home: we use a ``MoveTo`` stage and pass it the goal pose of ``ready``, which is a pose defined in the panda SRDF.
+
+.. code-block:: c++
+
+      {
+        auto stage = std::make_unique<mtc::stages::MoveTo>("return home", interpolation_planner);
+        stage->properties().configureInitFrom(mtc::Stage::PARENT, { "group" });
+        stage->setGoal("ready");
+        task.add(std::move(stage));
+      }
+
+All these stages should be added above these lines.
+
+.. code-block:: c++
+
+      // Stages all added to the task above this line
+
+      return task;
+    }
+
+
+Congratulations! You've now defined a pick and place task using MoveIt Task Constructor!
+
+Visualizing with RViz
+---------------------
+
+The task with each comprising stage is shown in the Motion Planning Tasks pane. Click on a stage and additional information about the stage will show up to the right. The right pane shows different solutions as well as their associated costs. Depending on the stage type and the robot configuration, there may only be one solution shown.
+
+Click a solution to see an animation of the robot following the plan for that stage. Click the "Exec" button in the upper-right portion of the pane to execute the motion.
```

Comment:
I've been confused by this in the past --- you need to click on the cost of a particular solution, then you can click Execute.

```suggestion
Click one of the solution costs to see an animation of the robot following the plan for that stage. Click the "Exec" button in the upper-right portion of the pane to execute the motion.
```